src/elements/content-explorer/ContentView.js

```diff
@@ -0,0 +1,119 @@
+// @flow
+
+import React from 'react';
+import AutoSizer from 'react-virtualized/dist/es/AutoSizer';
+import ItemList from './ItemList';
+import GridView from '../../components/grid-view/GridView';
+import type { ViewMode } from '../common/flowTypes';
+import { VIEW_MODE_LIST } from '../../constants';
+
+type Props = {
+    canDelete: boolean,
+    canDownload: boolean,
+    canPreview: boolean,
+    canRename: boolean,
+    canShare: boolean,
+    columnCount: number,
+    currentCollection: Collection,
+    focusedRow: number,
+    isMedium: boolean,
+    isSmall: boolean,
+    isTouch: boolean,
+    onItemClick: Function,
+    onItemDelete: Function,
+    onItemDownload: Function,
+    onItemPreview: Function,
+    onItemRename: Function,
+    onItemSelect: Function,
+    onItemShare: Function,
+    onSortChange: Function,
+    rootElement?: HTMLElement,
+    rootId: string,
+    slotRenderer: Function,
+    tableRef: Function,
+    view: View,
+    viewMode: ViewMode,
+};
+
+const ContentView = ({
+    canDelete,
+    canDownload,
+    canPreview,
+    canRename,
+    canShare,
+    columnCount,
+    currentCollection,
+    focusedRow,
+    isMedium,
+    isSmall,
+    isTouch,
+    onItemClick,
+    onItemDelete,
+    onItemDownload,
+    onItemPreview,
+    onItemRename,
+    onItemSelect,
+    onItemShare,
+    onSortChange,
+    rootElement,
+    rootId,
+    slotRenderer,
+    tableRef,
+    view,
+    viewMode,
```

Comment:
Consider making use of `{...rest}` to avoid repeating common/shared passed-through props both here and in the parent.